infra.bs

```diff
@@ -1074,35 +1081,99 @@ as 200/`<code>OK</code>`.
 
 <h2 id=json>JSON</h2>
 
+<p class=note>The conventions used in the algorithms in this section are those of the JavaScript
+specification. [[!ECMA-262]]
+
 <p>To <dfn export>parse JSON from bytes</dfn> given <var>bytes</var>, run these steps:
 
 <ol>
  <li><p>Let <var>jsonText</var> be the result of running <a>UTF-8 decode</a> on <var>bytes</var>.
  [[!ENCODING]]
 
- <li>
-  <p>Return ? <a abstract-op>Call</a>(<a>%JSONParse%</a>, undefined, « <var>jsonText</var> »).
-  [[!ECMA-262]]
-
-  <p class=note>The conventions used in this step are those of the JavaScript specification.
-
+ <li><p>Return ? <a abstract-op>Call</a>(<a>%JSONParse%</a>, undefined, « <var>jsonText</var> »).
 </ol>
 
 <p>To <dfn export>serialize JSON to bytes</dfn> a given JavaScript value <var>value</var>, run these
 steps:
 
 <ol>
  <li>
-  <p>Let <var>jsonString</var> be the result of
+  <p>Let <var>jsonString</var> be
   ? <a abstract-op>Call</a>(<a>%JSONStringify%</a>, undefined, « <var>value</var> »).
-  [[!ECMA-262]]
-  <p class=note>The conventions used in this step are those of the JavaScript specification.
-    Also, since no additional arguments are passed to <a>%JSONStringify%</a>, the resulting string
-    will have no whitespace inserted.
+
+  <p class=note>Since no additional arguments are passed to <a>%JSONStringify%</a>, the resulting
+  string will have no whitespace inserted.
 
  <li><p>Return the result of running <a>UTF-8 encode</a> on <var>jsonString</var>. [[!ENCODING]]
 </ol>
 
+<hr>
+
+<p>The above two operations operate on JavaScript values directly; in particular, this means that
+the involved objects or arrays are tied to a particular <a lt="realm">JavaScript realm</a>. In
+standards, it is often more convenient to parse JSON into realm-independent <a>maps</a>,
+<a>lists</a>, <a>strings</a>, <a>booleans</a>, numbers, and nulls.
+
+<p>To <dfn export>parse JSON into Infra values</dfn>, given a <a>string</a> <var>jsonText</var>:
+
+<ol>
+ <li><p>Let |jsValue| be ? [$Call$](<a>%JSONParse%</a>, undefined, « |jsonText| »).
```

Comment:
I wonder if it would be more useful for callers if this returned failure upon throwing.

Reply:
I think it's easier if callers can just re-throw the specific JSON parsing error, instead of having to remember that it's a SyntaxError, and using prose to stitch the error messages together. At least, that is what would work well for me in import maps.

Reply:
Okay, I was wondering if maybe returning failure would be better so you don't have to catch, but if you want to re-throw that makes sense.